hex-grid-generator.js

```diff
@@ -1,88 +1,84 @@
-var doc = app.activeDocument
-
-// Fill color for completed hexagons
-var fillColor = new RGBColor()
-fillColor.red = 230
-fillColor.blue = 60
-fillColor.green = 60
-
-// Stroke color for completed hexagons
-var strokeColor = new RGBColor()
-strokeColor.red = 30
-strokeColor.blue = 30
-strokeColor.green = 30
-
-// Is a single thing selected?
-if (doc.selection.length !== 1) {
-  // If not, panic and yell at them
-  var text = doc.textFrames.add()
-  text.move(doc, ElementPlacement.PLACEATBEGINNING)
-  text.contents = "Draw and select a single hexagon!\nHold down the shape tooland pick the\npolygon. Click and drag while holding down the shift key.\nRotate it 90 degrees if you want it pointy at the top."
-  text.left = 40
-  text.top = -100
-  text.textRange.characterAttributes.size = 30
-} else {
-  // If yes, we'll just assume it's a hexagon.
-  var hex = doc.selection[0]
-  var hexHeight = hex.height / 2
-  var hexWidth = hex.width / 2
-  var orientation = 'horizontal'
-  if (hexHeight > hexWidth) {
-    orientation = 'vertical'
+var doc = app.activeDocument;
+
+function IsSingleItemSelected(selection) {
+  return selection.length === 1;
+}
+
+function IsAHexagonSelected(selection) {
+  return selection[0].pathPoints.length === 6;
```

Comment:
Not absolutely necessary to check, but I wanted to try some of the other properties of that object.